### DISCUSSION TITLE
Add class to body when a menu is open

### DISCUSSION THREAD

**Discussion by CGTS** [alpinejs/alpine#2948] [1. Help] [answered]
I'm working on a menu with submenu and would like to add a class to the body (or anywhere) when the menu is open. I tried $store but my attempt was pretty sloppy and it broke the menu. Below is a simplified version of the menu I'm working on. Any suggestions on how to accomplish this would be appreciated.  

```
<body clas"class-when-menu-is-open">
<header 
  x-data="{ menuOpen: false }"
  class="">
  <div class="">
    <button 
      x-cloak
      @click="menuOpen = !menuOpen"
      class="">
      Main Trigger
    </button>
    <div 
      x-show="menuOpen"
      x-collapse.duration.500ms
      class="">
      <ul class="">
        <li 
          class="">
          Item 1
        </li>
        <li 
          x-data="{ menuOpen: false }" 
          class="">
          <button 
            @click.prevent="menuOpen = !menuOpen"
            @click.outside="menuOpen = false"
            class="">
            Sub Trigger
          </button>
          <div 
            x-cloak
            x-show="menuOpen"
            x-collapse.duration.500ms
            class="">
            Sub Content
          </div>
        </li>
      </ul>
    </div>
  </div>
</header>
</body>
```

**Comment by SimoTod** [comment, marked as answer]
There are multiple ways to do it (you can even put x-data on the body tag).
One option is
```html
<header 
  x-data="{ menuOpen: false }"
  x-effect="menuOpen ? document.body.classList.add('myClass') : document.body.classList.remove('test')"
>
```

**Reply by CGTS** [reply]
I appreciate this reply. You mentioned there are multiple ways. Is there a way to create a reusable function that can send a class to the body for different components. For example something I could attach to existing components like a menu, alert, or dropdown that will all send a class without having to add each on to the body.

**Reply by Tim-Wils** [reply]
You can also work with `$dispatch` in combination with `x-on`, but most of the time something like below can do the trick too:
```
<div class="comp" x-data="{ 
    opened: false,
    open() {
        this.opened= true;
        document.body.classList.add('comp-menu-open');
    },
    close() {
        this.opened= false;
        document.body.classList.remove('comp-menu-open');
    }
}"></div>
```
